### PR TITLE
Created browsing page

### PR DESCRIPTION
**Pull request by miyabitanimchi**
I created the browsing page. 
Now there is a warning like below, I already know why, it's because I put 2 components in the map method on `Browsing.tsx`. 

スクリーンショット 2021-12-15 午前2 04 37

Also I implemented a toggle function for the heart but it's not working yet because each card itself it's a link. (you can see the function works if you comment out the `Link` on the card)

And, for the layout of the cards, because I used `justify-center` and `flex-wrap` the card(s) on a last row goes in the center. Probably it's better it(they) are placed in the left side right?

Let me figure out how we can improve them later 🙏

### CODE REVIEW

**Comment by shiv-chan** [comment]
Thank you @miyabitanimchi for your hard work!
I'm gonna review the code and test it on my browser.

**Just leave a quick comment.**
- I guess the error message shows up because there is no `key` for maps not because of having two maps according to the message.
- what if place the heart toggle with `absolute` or set `z-index` to make it placed above the card itself? Probably, "event propagation" will happen there.
- why don't you use `grid` for this layout instead?

**Comment by miyabitanimchi** [comment]
What I changed and fixed are:

- For the layout, I changed Flexbox to Grid.
- On `Browsing.tsx` I added a function to create a new array to extract data for senpai and kohai, and map the array to arrange the user cards (no warning anymore)
- And now you can toggle the heart! 
- Added "... and more" if there are more than 5 skill tags

Thank you for your advice @shiv-chan 💡
Please review again when you can, thank you!

File: client/src/common/components/browsingPage/UserCard.tsx [hunk 1]
```diff
@@ -27,32 +27,39 @@ const UserCard = ({
       return name;
     }
   };
+
+  const handleLike = (e: React.MouseEvent<SVGAElement>) => {
+    e.preventDefault();
+    setIsLiked((prevState) => !prevState);
+    e.stopPropagation();
```
Comment (shiv-chan) [inline client/src/common/components/browsingPage/UserCard.tsx:34]: yay 🎉 🎉  now it's working! Thanks!

File: client/src/common/components/browsingPage/UserCard.tsx [hunk 2]
```diff
@@ -27,32 +27,39 @@ const UserCard = ({
...
+
   return (
     <Link
       to={`/profile/${isSenpai ? 'senpai' : 'kohai'}/${profile.id}`}
-      className="z-0"
+      className="z-0 w-full h-full"
     >
-      <div className="mobile_l_max:h-48 mobile_l_max:w-72 mobile_l_max:p-2.5 h-56 w-80 mobile_l:h-64 mobile_l:w-96 flex-initial bg-white hover:bg-gray-50 rounded-lg p-5 z-0">
-        <div className="flex justify-between items-start mb-7 mb-l-max:mb-2">
+      <div className="w-full h-60 mobile_l_max:h-48 flex-initial bg-white hover:bg-gray-50 rounded-lg mobile_l_max:p-2.5 p-5 z-0">
```
Comment (shiv-chan) [inline client/src/common/components/browsingPage/UserCard.tsx:42]: @miyabitanimchi I liked the colour slightly changes when hovering 💖

**Review by shiv-chan — APPROVED** [review]
Thanks a million, @miyabitanimchi 🙏 
The layout and the code are much nicer and cleaner!

Let me make a comment about the spacing on mobile devices.
As you see in the capture below, each card takes up almost all width of the screen, so looks tight. 
スクリーンショット 2021-12-15 22 12 14

I updated this in `feature-browsing-page-kaho` branch and it looks like the below.
スクリーンショット 2021-12-16 0 10 25

I'm going to make a PR to merge this branch, so please take a look.

Thanks again!

**Pull request merged**